### PR TITLE
git: update to 2.8.3

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

File: components/developer/git/Makefile [hunk 1]
```diff
@@ -23,13 +23,12 @@
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		git
-COMPONENT_VERSION=	1.9.4
-COMPONENT_REVISION=	3
+COMPONENT_VERSION=	2.8.1
```
Comment (avsej) [inline components/developer/git/Makefile]: @pyhalov, solaris already using 2.7.4

**Comment by pyhalov** [comment]
need to find out if webrev and other onbld tools like new git

**Comment by pyhalov** [comment]
Is there any analysis why tests are failing?

**Comment by avsej** [comment]
No, I only identified failing tests and commented them out in patches

**Comment by avsej** [comment]
@pyhalov I suspect that it might be because it links with iconv from libc, and not GNU libiconv, but I didn't managed to compile with GNU libiconv to prove that.

**Comment by alarcher** [comment]
Configure complains mildly about grep not being GNU grep so you may need to prepend /usr/gnu/bin to PATH.

**Comment by alarcher** [comment]
With --with-iconv and adding -I/usr/gnu/include the disabled tests pass but still:

```
# failed 4 among remaining 91 test(s)

not ok 19 - -p handles "no changes" gracefully
#   
#       HEAD=$(git rev-parse HEAD) &&
#       set_fake_editor &&
#       git rebase -i -p HEAD^ &&
#       git update-index --refresh &&
#       git diff-files --quiet &&
#       git diff-index --quiet --cached HEAD -- &&
#       test $HEAD = $(git rev-parse HEAD)
#   

not ok 21 - preserve merges with -p
#   
#       git checkout -b to-be-preserved master^ &&
#       : > unrelated-file &&
#       git add unrelated-file &&
#       test_tick &&
#       git commit -m "unrelated" &&
#       git checkout -b another-branch master &&
#       echo B > file1 &&
#       test_tick &&
#       git commit -m J file1 &&
#       test_tick &&
#       git merge to-be-preserved &&
#       echo C > file1 &&
#       test_tick &&
#       git commit -m K file1 &&
#       echo D > file1 &&
#       test_tick &&
#       git commit -m L1 file1 &&
#       git checkout HEAD^ &&
#       echo 1 > unrelated-file &&
#       test_tick &&
#       git commit -m L2 unrelated-file &&
#       test_tick &&
#       git merge another-branch &&
#       echo E > file1 &&
#       test_tick &&
#       git commit -m M file1 &&
#       git checkout -b to-be-rebased &&
#       test_tick &&
#       set_fake_editor &&
#       git rebase -i -p --onto branch1 master &&
#       git update-index --refresh &&
#       git diff-files --quiet &&
#       git diff-index --quiet --cached HEAD -- &&
#       test $(git rev-parse HEAD~6) = $(git rev-parse branch1) &&
#       test $(git rev-parse HEAD~4^2) = $(git rev-parse to-be-preserved) &&
#       test $(git rev-parse HEAD^^2^) = $(git rev-parse HEAD^^^) &&
#       test $(git show HEAD~5:file1) = B &&
#       test $(git show HEAD~3:file1) = C &&
#       test $(git show HEAD:file1) = E &&
#       test $(git show HEAD:unrelated-file) = 1
#   
not ok 22 - edit ancestor with -p
#   
#       set_fake_editor &&
#       FAKE_LINES="1 2 edit 3 4" git rebase -i -p HEAD~3 &&
#       echo 2 > unrelated-file &&
#       test_tick &&
#       git commit -m L2-modified --amend unrelated-file &&
#       git rebase --continue &&
#       git update-index --refresh &&
#       git diff-files --quiet &&
#       git diff-index --quiet --cached HEAD -- &&
#       test $(git show HEAD:unrelated-file) = 2
#   
not ok 23 - --continue tries to commit
#   
#       test_tick &&
#       set_fake_editor &&
#       test_must_fail git rebase -i --onto new-branch1 HEAD^ &&
#       echo resolved > file1 &&
#       git add file1 &&
#       FAKE_COMMIT_MESSAGE="chouette!" git rebase --continue &&
#       test $(git rev-parse HEAD^) = $(git rev-parse new-branch1) &&
#       git show HEAD | grep chouette
#   


```

**Comment by alarcher** [comment]
Setting PATH=/usr/gnu/bin:/usr/bin and ENV=/usr/bin/env -i passes all the tests except Japanese encoding.
Failures are false positive due to wrong environment during gmake test.
Tested with 2.8.3

**Comment by pyhalov** [comment]
Can we update PR and set PATH accordingly in test target?

**Comment by avsej** [comment]
I've update test environment, but still some of them fail

```
ok 41 - setup hunk header tests
not ok 42 - hunk header: README
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 README >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 43 - hunk header: cpp-c++-function
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-c++-function >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 44 - hunk header: cpp-class-constructor
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-class-constructor >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 45 - hunk header: cpp-class-constructor-mem-init
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-class-constructor-mem-init >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 46 - hunk header: cpp-class-definition
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-class-definition >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 47 - hunk header: cpp-class-definition-derived
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-class-definition-derived >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 48 - hunk header: cpp-class-destructor
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-class-destructor >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 49 - hunk header: cpp-function-returning-global-type
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-function-returning-global-type >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 50 - hunk header: cpp-function-returning-nested
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-function-returning-nested >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 51 - hunk header: cpp-function-returning-pointer
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-function-returning-pointer >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 52 - hunk header: cpp-function-returning-reference
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-function-returning-reference >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 53 - hunk header: cpp-gnu-style-function
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-gnu-style-function >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 54 - hunk header: cpp-namespace-definition
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-namespace-definition >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 55 - hunk header: cpp-operator-definition
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-operator-definition >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 56 - hunk header: cpp-skip-access-specifiers
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-skip-access-specifiers >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 57 - hunk header: cpp-skip-comment-block
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-skip-comment-block >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 58 - hunk header: cpp-skip-labels
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-skip-labels >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 59 - hunk header: cpp-struct-definition
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-struct-definition >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 60 - hunk header: cpp-struct-single-line
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-struct-single-line >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 61 - hunk header: cpp-template-function-definition
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-template-function-definition >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 62 - hunk header: cpp-union-definition
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-union-definition >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 63 - hunk header: cpp-void-c-function
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 cpp-void-c-function >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 64 - hunk header: custom1-pattern
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 custom1-pattern >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 65 - hunk header: custom2-match-to-end-of-line
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 custom2-match-to-end-of-line >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 66 - hunk header: custom3-alternation-in-pattern
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 custom3-alternation-in-pattern >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 67 - hunk header: fountain-scene
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 fountain-scene >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 68 - hunk header: java-class-member-function
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 java-class-member-function >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 69 - hunk header: perl-skip-end-of-heredoc
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 perl-skip-end-of-heredoc >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 70 - hunk header: perl-skip-forward-decl
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 perl-skip-forward-decl >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 71 - hunk header: perl-skip-sub-in-pod
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 perl-skip-sub-in-pod >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 72 - hunk header: perl-sub-definition
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 perl-sub-definition >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
not ok 73 - hunk header: perl-sub-definition-kr-brace
#   
#           test_when_finished 'cat actual' &&  # for debugging only
#           git diff -U1 perl-sub-definition-kr-brace >actual &&
#           grep '@@ .* @@.*RIGHT' actual
#       
# failed 32 among 73 test(s)
1..73
Makefile:43: recipe for target 't4018-diff-funcname.sh' failed
```

**Comment by avsej** [comment]
@alarcher @pyhalov now all tests are passing, only japanese-related marked as likely broken, but still running.

**Comment by pyhalov** [comment]
does your build system have pkg:/locale/ja installed?

**Comment by alarcher** [comment]
Yes

**Comment by avsej** [comment]
installing `pkg:/locale/ja` does not help, those tests still failing

**Comment by pyhalov** [comment]
OK, please merge commits into one and let's be done with it

File: components/developer/git/Makefile [hunk 2]
```diff
@@ -97,7 +108,20 @@ $(BUILD_DIR)/.manpages:
...
+$(BUILD_DIR_32)/.bash-completion:
+	$(MKDIR) $(PROTOUSRSHAREDIR)/bash-completion/completions
+	$(CP) -RLp $(BUILD_DIR_32)/contrib/completion/git-completion.bash $(PROTOUSRSHAREDIR)/bash-completion/completions/git
+	(cd $(PROTOUSRSHAREDIR)/bash-completion/completions; $(LN) -s git gitk)
+	$(TOUCH) $@
 
-BUILD_PKG_DEPENDENCIES =	$(BUILD_TOOLS)
-
-include $(WS_TOP)/make-rules/depend.mk
+include $(WS_MAKE_RULES)/depend.mk
```
Comment (pyhalov) [inline components/developer/git/Makefile]: this include is not needed and should be removed

File: components/developer/git/Makefile [hunk 3]
```diff
@@ -23,52 +23,63 @@
...
 # Therefore we need cloney to copy a set of files to build.
+# Overwrite symliked tests, because they relying on test assets being regular files.
 
 COMPONENT_PRE_CONFIGURE_ACTION = \
-	(chmod u+x $(COMPONENT_SRC)/t/*.sh  && $(CLONEY) $(SOURCE_DIR) $(@D))
+	(chmod u+x $(COMPONENT_SRC)/t/*.sh  && $(CLONEY) $(SOURCE_DIR) $(@D) && \
+	$(RM) -r $(@D)/t && $(CP) -rp $(COMPONENT_SRC)/t $(@D))
+
+
+ASLR_MODE = $(ASLR_ENABLE)
```
Comment (pyhalov) [inline components/developer/git/Makefile]: we currently don't support this, but better remove it, as if it is enabled one day, it should be tested

File: components/developer/git/git.p5m [hunk 4]
```diff
@@ -274,21 +228,23 @@ file path=usr/lib/git-core/mergetools/tkdiff
...
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/Editor.pm
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/Fetcher.pm
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/GlobSpec.pm
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/Log.pm
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/Memoize/YAML.pm
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/Migration.pm
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/Prompt.pm
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/Ra.pm
-file path=usr/perl5/vendor_perl/$(PERL_VERSION)/i86pc-solaris-64int/Git/SVN/Utils.pm
+file path=usr/perl5/vendor_perl/5.22/i86pc-solaris-64int/Error.pm
```
Comment (pyhalov) [inline components/developer/git/git.p5m]: Why $(PERL_VERSION) didn't work?

Comment (avsej) [inline components/developer/git/git.p5m]: it was generated by `sample-manifest`

**Pull request merged**